server/src/main/java/org/elasticsearch/search/SearchHits.java

```diff
@@ -252,6 +252,7 @@ public boolean decRef() {
     }
 
     private void deallocate() {
+        var hits = this.hits;
```

Comment:
Just saving the memory reference, turns out this was what made this method show up in profiling.